src/setup/helpers.ts

```diff
@@ -107,3 +108,38 @@ export function hashSignedObj(obj): string {
   }
   return crypto.hashObj(obj, true)
 }
+
+/**
+@param rawPayload: any - The original payload stripped of the signatures
+@param sigs: Sign[] - The signatures to verify
+@param allowedPubkeys: {[pubkey: string]: DevSecurityLevel} - The public keys that are allowed to sign the payload
+@param minSigRequired: number - The minimum number of signatures required
+@param requiredSecurityLevel: DevSecurityLevel - The minimum security level required to sign the payload
+@returns boolean - True if the payload is signed by the required number of authorized public keys with the required security level
+**/
+export function verifyMultiSigs(
+  rawPayload: {}, 
+  sigs: Sign[], 
+  allowedPubkeys: {[pubkey: string]: DevSecurityLevel}, 
+  minSigRequired: number, 
+  requiredSecurityLevel: DevSecurityLevel
+): boolean {
+
+  let validSigs = 0
+  const payload_hash = crypto.hashObj(rawPayload)
+  for(let i = 0; i < sigs.length; i++){
+
+    // The sig owner is listed on the server and
+    // The sig owner has enough security clearance
+    // The signature is valid
+    if(
+        allowedPubkeys[sigs[i].owner] && 
+        allowedPubkeys[sigs[i].owner] >= requiredSecurityLevel && 
+        crypto.verify(payload_hash, sigs[i].sig, sigs[i].owner)
+    ){
+      validSigs++
```

Comment:
If we reach minSigRequired we can exit this loop early.

Reply:
This is done